### PR TITLE
Database: Quests, Shop and ShopItems

### PR DESCRIPTION
**Pull request by AngeloTadeucci**
- Removed `ShopParser` and added `Shops` and `ShopItems` to database;
- Seeding Shops after database is created;
- Added `Quests` to database;
- Changes to `Guild`, using the database to generate uids instead of `UidGenerator`.

### CODE REVIEW

File: MapleServer2/Types/QuestStatus.cs [hunk 1]
```diff
@@ -1,33 +1,72 @@
...
+            QuestMetadata metadata = QuestMetadataStorage.GetMetadata(id);
+            Basic = metadata.Basic;
+            StartNpcId = metadata.StartNpc;
+            CompleteNpcId = metadata.CompleteNpc;
+            Reward = metadata.Reward;
+            RewardItems = metadata.RewardItem;
+        }
+    }
+
+    public class Condition
```
Comment (AlanMorel) [inline MapleServer2/Types/QuestStatus.cs:55]: New class?

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/QuestStatus.cs (hunk 1 above)
Comment (AngeloTadeucci) [reply MapleServer2/Types/QuestStatus.cs:55]: Didn't want to change values inside the metadata class. If you feel it's ok I can revert it to keep using the metadata class

File: MapleServer2/Database/Types/Shop.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        public bool RestrictSales { get; set; }
+        public bool CanRestock { get; set; }
+        public long NextRestock { get; set; }
+        public bool AllowBuyback { get; set; }
+        public List<ShopItem> Items { get; set; }
+
+        public Shop() { }
+    }
+
+    public class ShopItem
```
[the remark this answers is not in the document]
Comment (AngeloTadeucci) [reply MapleServer2/Database/Types/Shop.cs:23]: Sure

File: MapleServer2/Types/QuestStatus.cs (hunk 1 above)
Comment (AlanMorel) [reply MapleServer2/Types/QuestStatus.cs:55]: Oh gotcha, nevermind

**Review by AlanMorel — APPROVED** [review]

**Pull request merged**